python/lsst/cp/pipe/defects.py

```diff
@@ -0,0 +1,645 @@
+# This file is part of cp_pipe.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+__all__ = ['FindDefectsTask',
+           'FindDefectsTaskConfig', ]
+
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.backends.backend_pdf import PdfPages
+import os
+import warnings
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.afw.image as afwImage
+import lsst.meas.algorithms as measAlg
+import lsst.afw.math as afwMath
+import lsst.afw.detection as afwDetection
+import lsst.log as lsstLog
+import lsst.afw.display as afwDisplay
+import lsst.afw.cameraGeom.utils as cgUtils
+
+from lsst.ip.isr import IsrTask
+from .utils import NonexistentDatasetTaskDataIdContainer, SingleVisitListTaskRunner, countMaskedPixels, \
+    validateIsrConfig
+
+
+class FindDefectsTaskConfig(pexConfig.Config):
+    """Config class for defect finding"""
+
+    isrForFlats = pexConfig.ConfigurableField(
+        target=IsrTask,
+        doc="""Task to perform instrumental signature removal""",
+    )
+    isrForDarks = pexConfig.ConfigurableField(
+        target=IsrTask,
+        doc="""Task to perform instrumental signature removal""",
```

Comment:
I think, for some use-cases, it might be reasonable to support doing a dark correction on a flat, but that would never be the case on a dark exposure, as those would only ever be being used to find the bright defects, so I just wanted to make sure that both exposure types could be supported by the same running of the task, but with different isr configs.